components/layout/display_list_builder.rs

```diff
@@ -1165,17 +1165,24 @@ impl FragmentDisplayListBuilding for Fragment {
         let size = unbordered_box.size.to_physical(style.writing_mode);
         let name = paint_worklet.name.clone();
 
-        // If the script thread has not added any paint worklet modules, there is nothing to do!
-        let executor = match state.layout_context.paint_worklet_executor {
-            Some(ref executor) => executor,
-            None => return debug!("Worklet {} called before any paint modules are added.", name),
+        // Get the painter, and the computed values for its properties.
+        let (properties, painter) = match state.layout_context.registered_painters.read().get(&name) {
+            Some(registered_painter) => (
+                registered_painter.properties
+                    .iter()
+                    .filter_map(|(name, id)| id.as_shorthand().err().map(|id| (name, id)))
```

Comment:
Yes, but very annoyingly shorthand property ids have a lifetime, so we can't store them in the properties map.